pycbc/types/array_cupy.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (C) 2012  Alex Nitz
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+"""Numpy based CPU backend for PyCBC Array
+"""
+import cupy as _xp
```

Comment:
Not a problem, just wondering why `_xp` over `_cp`?

Reply:
In BBHx `_xp` is used to indicate "numpy or cupy, we don't know or care here". In this module it should probably be `cp` sp this is changed.